test/test_broker/iaas.AliClient.spec.js

```diff
@@ -0,0 +1,295 @@
+    } else if (file === invalidBlobName2) {
+      return Promise.reject(new Forbidden(`Authentication at ali cloud storage provider failed while deleting blob ${file} in container ${settings.container}`));
+    } else if (file === notFoundBlobName) {
+      return Promise.reject(new NotFound(`Object '${file}' not found while deleting in container ${settings.container}`));
+    }
+  }
+};
+describe('iaas', function () {
+  describe('AliClient', function () {
+    describe('#AliStorage', function () {
+      it('should form an object with correct credentials', function () {
+        const responseAliStorageObject = AliClient.createStorageClient(settings);
+        expect(responseAliStorageObject.options.accessKeyId).to.equal(settings.keyId);
+        expect(responseAliStorageObject.options.accessKeySecret).to.equal(settings.key);
+        expect(responseAliStorageObject.options.region).to.equal(settings.region);
+        expect(responseAliStorageObject.options.endpoint.href).to.equal(settings.endpoint + '/');
+        expect(responseAliStorageObject.options.endpoint.hostname).to.equal(_.split(settings.endpoint, '//')[1]);
+        expect(responseAliStorageObject.options.endpoint.protocol).to.equal(_.split(settings.endpoint, '//')[0]);
+      });
+    });
+
+    describe('#BucketOperations', function () {
+      let sandbox, client;
+      beforeEach(function () {
+        sandbox = sinon.createSandbox();
+        client = new AliClient(settings);
+        sandbox.stub(AliStorage.prototype, 'listBuckets').withArgs({ prefix: settings.container }).callsFake(bucketStub);
+        sandbox.stub(AliStorage.prototype, 'useBucket').withArgs(settings.container).returns(listFilesStub);
+      });
+      afterEach(function () {
+        sandbox.restore();
+      });
+
+      it('container properties should be retrived successfully', function () {
+        return client.getContainer()
+          .then(result => {
+            expect(result[0].name).to.equal(settings.container);
+            expect(result[0].id).to.equal(settings.container);
+            expect(result[0].location).to.equal(settings.region);
+          })
+          .catch(err => {
+            logger.error(err);
+            console.log(err);
+            throw new Error('expected container properties to be retrived successfully');
+          });
+      });
+
+      it('getting container properties should fail with Not Found error', function () {
+        sandbox.restore();
+        sandbox.stub(AliStorage.prototype, 'useBucket').withArgs(settings.container).returns(listFilesStub);
+        sandbox.stub(AliStorage.prototype, 'listBuckets').withArgs({ prefix: settings.container }).callsFake(incorrectBucketStub);
+        return client.getContainer()
+          .then(() => {
+            logger.error('The get container call should fail');
```

Comment:
In this case, it is expected to fail the test here. I think it will be missed by just logging error.

Reply:
fixed!